server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueue.java

```diff
@@ -119,4 +123,11 @@ public String toString() {
             .add("name", name)
             .toString();
     }
+
+    public Mono<Integer> republishNotProcessedMails(Instant olderThan) {
```

Comment:
With such an API you only have the reporting at the end of the task.

What is interesting with the count of republish items is to have it updated in live.

How about letting the caller specify a mutable context, incremented on each requeued elements? The additional information can then rely on this context to expose the progress.

This is comonly how we handle task progress reporting...